posydon/visualization/plot_pop.py

```diff
@@ -251,4 +258,137 @@ def plot_hist_properties(x, df_intrinsic=None, df_observable=None,
     if path:
         plt.savefig(path)
     if show:
-        plt.show()
\ No newline at end of file
+        plt.show()
+        
+        
+def plot_popsyn_over_grid_slice(pop, grid_type, met_Zsun, slices=None, channel=None, 
+                                plot_dir='./', prop=None, prop_range=None, 
+                                log_prop=False, alpha=0.3, s=5., 
+                                show_fig=True, save_fig=True, close_fig=True,
+                                verbose=False):
+
+    # load grid
+    met = convert_metallicity_to_string(met_Zsun)
+    grid_path = os.path.join(PATH_TO_POSYDON_DATA, f'POSYDON_data/{grid_type}/{met}_Zsun.h5')
+    grid = PSyGrid(verbose=False)
+    grid.load(grid_path)
+
+    # check if formation channel information is avaialbe
+    if channel is not None and 'channel' not in pop.df_oneline.keys():
+        if verbose:
+            print('Computing formation channels...')
+        pop.get_formation_channels()
+        
+    if 'CO' in grid_path:
+        # compact object mass slices
+        def log_range(x_min,x_max,x_n):
+            return 10**np.linspace(np.log10(x_min),np.log10(x_max), x_n)
+        m_CO_min = 1.
+        m_CO_max = 51.32759630397827
+        m_CO_n = 23
+        m_COs = log_range(m_CO_min, m_CO_max, m_CO_n)
+        m_COs_edges = 10**(np.log10(np.array(m_COs)[:-1])+
+                                (np.log10(np.array(m_COs)[1:])-np.log10(np.array(m_COs)[:-1]))/2)
+        m2 = [0.]+m_COs_edges.tolist()+[55.]
+        vars = m_COs
+        fname = 'grid_m_%1.2f.png'
+        title = '$m_\mathrm{CO}=%1.2f\,M_\odot$'
+        slice_3D_var_str='star_2_mass'
+    elif 'HMS-HMS' in grid_path:
+        # mass ratio slices
+        qs = np.linspace(0.05,1.,20)
+        qs[-1] = 0.99
+        vars = qs.tolist()
+        fname = 'grid_q_%1.2f.png'
+        title = '$q=%1.2f$'
+        slice_3D_var_str='mass_ratio'
+    else:
+        raise ValueError('Grid type not supported!')
+
+
+    for i, var in enumerate(vars):
+        if slices is not None and round(var,2) not in np.around(slices,2):
+            if verbose:
+                print(f'Skipping {round(var,2)}')
+            continue
+        if 'HMS-HMS' in grid_path:
+            slice_3D_var_range = (var-2.5e-2,var+2.5e-2)
```

Comment:
This don't need to be hard coded as `2.5e-2` either. By the way, this is already questionable in the case of q=0.99, which would now get an overlap with q=0.95.

Reply:
Fixed, I did it this because on the moment it was annoying to write down the lines of code that compute the ranges dinamically as you need to convert from from array to string to do the manipulation and then manyally add the edges.